thalerj/openbmctool.py

```diff
@@ -0,0 +1,1490 @@
+        bmcState = json.loads(res.text)['data'].split('.')[-1]
+        if(args.json):
+            outDict = {"Chassis Power State" : chassisState, "Host Power State" : hostState, "BMC Power State":bmcState}
+            return json.dumps(outDict, sort_keys=True, indent=4, separators=(',', ': '), ensure_ascii=False)
+        else:
+            return "Chassis Power State: " +chassisState + "\nHost Power State: " + hostState + "\nBMC Power State: " + bmcState
+    else:
+        return "Invalid chassis power command"
+
+"""
+     called by the chassis function. Controls the identify led of the chassis. Sets or gets the state
+       
+     @param host: string, the hostname or IP address of the bmc
+     @param args: contains additional arguments used by the fru sub command
+     @param session: the active session to use
+     @param args.json: boolean, if this flag is set to true, the output will be provided in json format for programmatic consumption 
+"""
+def chassisIdent(host, args, session):
+    if(args.identcmd == 'on'):
+        print("Attempting to turn identify light on...:")
+        url="https://"+host+"/xyz/openbmc_project/led/groups/enclosure_identify/attr/Asserted"
+        httpHeader = {'Content-Type':'application/json',}
+        data = '{"data":true}'
+        res = session.put(url, headers=httpHeader, data=data, verify=False, timeout=20)
+        return res.text
+    elif(args.identcmd == 'off'):
+        print("Attempting to turn identify light off...:")
+        url="https://"+host+"/xyz/openbmc_project/led/groups/enclosure_identify/attr/Asserted"
+        httpHeader = {'Content-Type':'application/json'}
+        data = '{"data":false}'
+        res = session.put(url, headers=httpHeader, data=data, verify=False, timeout=20)
+        return res.text
+    elif(args.identcmd == 'status'):
+        url="https://"+host+"/xyz/openbmc_project/led/groups/enclosure_identify"
+        httpHeader = {'Content-Type':'application/json'}
+#         print(url)
+        res = session.get(url, headers=httpHeader, verify=False, timeout=20)
+        status = json.loads(res.text)['data']
+        if(args.json):
+            return status
+        else:
+            if status['Asserted'] == 0:
+                return "Identify light is off"
+            else:
+                return "Identify light is blinking"
+    else:
+        return "Invalid chassis identify command"
+
+"""
+     controls the different chassis commands
+       
+     @param host: string, the hostname or IP address of the bmc
+     @param args: contains additional arguments used by the fru sub command
+     @param session: the active session to use
+     @param args.json: boolean, if this flag is set to true, the output will be provided in json format for programmatic consumption 
+""" 
+def chassis(host, args, session):
+    if(hasattr(args, 'powcmd')):
+        result = chassisPower(host,args,session)
```

Comment:
Can't we use argparse to do the right thing here rather than dispatching in the helper like this?